components/script_bindings/interfaces.rs

```diff
@@ -23,6 +23,15 @@ use crate::script_runtime::{CanGc, JSContext};
 use crate::settings_stack::StackEntry;
 use crate::utils::ProtoOrIfaceArray;
 
+/// Operations that can be invoked for a WebIDL interface against
+/// a global object.
+pub struct Interface {
+    /// Define the JS object for this interface on the given global.
+    pub define: fn(JSContext, HandleObject),
+    /// Returns true if this interface's conditions are met for the given global.
+    pub enabled: fn(JSContext, HandleObject) -> bool,
```

Comment:
I would also add link here to mozilla: https://searchfox.org/mozilla-central/rev/2c71f1e9b5947612abdc16b64008162c58c1b9d3/dom/bindings/WebIDLGlobalNameHash.h#35